state/state.go

```diff
@@ -118,10 +118,7 @@ func (state State) MakeBlock(
 
 	// Set time
 	if height == 1 {
-		block.Time = tmtime.Now()
-		if block.Time.Before(state.LastBlockTime) {
-			block.Time = state.LastBlockTime // state.LastBlockTime for height == 1 is genesis time
-		}
+		block.Time = state.LastBlockTime // genesis time
```

Comment:
This modification lead to generation of the same blocks for height 1, which leads to some consensus tests failing as they assumed that whenever MakeBlock is created, it will return new block. As for height 1 we don't have a way to constrain time, probably it is probably acceptable behaviour to have idempotent behaviour for MakeBlock. We just need to fix failing tests to modify returned block in case new block is needed.

Reply:
Thank you 🍏 

Why don't add `round` to block's header? Or we could start from `height=2` as an alternative solution.

Reply:
Maybe better option is to modify failing test. For example https://github.com/tendermint/tendermint/pull/2607.